modo/api.py

```diff
@@ -125,17 +134,28 @@ def show_contents(self):
 
     def list_files(self) -> Generator[Path, None, None]:
         """Lists files in the archive recursively (except for the zarr file)."""
-        for path in self.path.glob("*"):
-            if path.name.endswith(".zarr"):
-                continue
-            elif path.is_file():
-                yield path
-            for file in path.rglob("*"):
-                yield file
+        if isinstance(self.archive.store, zarr.storage.FSStore):
+            fs = self.archive.store.fs
+            for path in fs.find(str(self.path)):
+                if Path(path).name.endswith(".zarr") or Path(
+                    path
+                ).name.startswith("."):
```

Comment:
**clarification**: is there a reason why we're excluding dot files on the remote and not on local?

Reply:
Very good point and well spotted! The reason was in the different return of `fs.find(self.path)` (returns all files at file level) and `self.path.glob("*")` (returns the files and subdirectories of path). So this part `Path(path).name.endswith(".zarr")` did not properly exclude the zarr archive. Excluding dot files was a dirty fix and could indeed result in a different behavior between local and remote listing. In 1ab950d699cf0756bb89e019fa56da187c611281 I changed the conditionals to explicitly exclude the zarr archive and list all files and related files in subdirectories.